src/host/alias.cpp

```diff
@@ -818,350 +822,134 @@ void Alias::s_ClearCmdExeAliases()
-size_t Alias::s_ReplaceMacros(std::wstring& str,
-                              const std::deque<std::wstring>& tokens,
-                              const std::wstring& fullArgString)
-{
-    size_t lineCount = 0;
-    std::wstring finalText;
+    std::wstring buffer;
+    size_t lines = 0;
 
-    // The target text may contain substitution macros indicated by $.
-    // Walk through and substitute them as appropriate.
-    for (auto ch = str.cbegin(); ch < str.cend(); ch++)
+    // Put together the target string.
+    for (auto it = target.begin(), end = target.end(); it != end;)
     {
-        if (L'$' == *ch)
+        auto ch = *it++;
+        if (ch != L'$' || it == end)
         {
-            // Attempt to read ahead by one character.
-            const auto chNext = ch + 1;
+            buffer.push_back(ch);
+            continue;
+        }
 
-            if (chNext < str.cend())
+        ch = *it++;
+        const auto chLower = til::tolower_ascii(ch);
+        if (chLower >= L'1' && chLower <= L'9')
+        {
+            // numbered parameter substitution
+            const size_t ArgNumber = chLower - L'1';
+            if (ArgNumber < argsCount)
             {
-                auto isProcessed = s_TryReplaceNumberedArgMacro(*chNext, finalText, tokens);
-                if (!isProcessed)
-                {
-                    isProcessed = s_TryReplaceWildcardArgMacro(*chNext, finalText, fullArgString);
-                }
-                if (!isProcessed)
-                {
-                    isProcessed = s_TryReplaceInputRedirMacro(*chNext, finalText);
-                }
-                if (!isProcessed)
-                {
-                    isProcessed = s_TryReplaceOutputRedirMacro(*chNext, finalText);
-                }
-                if (!isProcessed)
-                {
-                    isProcessed = s_TryReplacePipeRedirMacro(*chNext, finalText);
-                }
-                if (!isProcessed)
-                {
-                    isProcessed = s_TryReplaceNextCommandMacro(*chNext, finalText, lineCount);
-                }
-                if (!isProcessed)
-                {
-                    // If nothing matches, just push these two characters in.
-                    finalText.push_back(*ch);
-                    finalText.push_back(*chNext);
-                }
-
-                // Since we read ahead and used that character,
-                // advance the iterator one extra to compensate.
-                ch++;
+                buffer.append(args[ArgNumber]);
             }
-            else
+        }
+        else if (chLower == L'*')
+        {
+            // * parameter substitution = all parameters
+            if (argsCount)
             {
-                // If no read-ahead, just push this character and be done.
-                finalText.push_back(*ch);
+                buffer.append(args[0].data(), sourceText.data() + sourceText.size());

```

Comment:
I think this deserves a comment. We're sure that `args` contains views into `sourceText`, so even though it _looks like_ we're using a `begin` from one pointer and an `end` from a totally different one, it's OK

plus the understanding that `args` is _only the arguments_ (e.g. it doesn't work like `argv` where `[0]` is the program name (!)

Reply:
Done. Also it now works like `argv`.